tests/typecheck/managers/test_managers.yml

```diff
@@ -386,9 +386,11 @@
 -   case: override_manager_create1
     main: |
         from myapp.models import MyModel
-        MyModel.objects.create()
     installed_apps:
         - myapp
+    out: |
+        myapp/models:4: error: Return type "MyModel" of "create" incompatible with return type "_T" in supertype "BaseManager"
+        myapp/models:5: error: Incompatible return value type (got "_T", expected "MyModel")
```

Comment:
This is the failing test I mentioned in the PR description. I decided to fix it like this, basically saying this is an error. I think it should be fine to raise an error if you've restricted the type of an overridden method but the type vars of to the manager class